runtime/vm/extendedMessageNPE.cpp

```diff
@@ -159,19 +374,24 @@ getCompleteNPEMessage(J9VMThread *vmThread, U_8 *bcCurrentPtr, J9ROMClass *romCl
 			Trc_VM_GetCompleteNPEMessage_Invoke_Index(vmThread, index);
 
 			J9ROMMethodRef *romMethodRef = (J9ROMMethodRef *)&constantPool[index];
-			J9ROMNameAndSignature *nameAndSig = J9ROMMETHODREF_NAMEANDSIGNATURE(romMethodRef);
-			J9UTF8 *sig = J9ROMNAMEANDSIGNATURE_SIGNATURE(nameAndSig);
-			J9UTF8 *name = J9ROMNAMEANDSIGNATURE_NAME(nameAndSig);
-			const char *VM_NPE_INVOKEMETHOD = "Cannot invoke \"%2$.*1$s.%4$.*3$s%6$.*5$s\"";
-			J9UTF8 *definingUTF = J9ROMSTRINGREF_UTF8DATA((J9ROMStringRef *)&constantPool[romMethodRef->classRefCPIndex]);
+			J9ROMNameAndSignature *methodNameAndSig = J9ROMMETHODREF_NAMEANDSIGNATURE(romMethodRef);
+			J9UTF8 *methodName = J9ROMNAMEANDSIGNATURE_NAME(methodNameAndSig);
+			const char *VM_NPE_INVOKEMETHOD = "Cannot invoke \"%s.%.*s%s\"";
+
+			J9UTF8 *definingClassFullQualifiedName = J9ROMSTRINGREF_UTF8DATA((J9ROMStringRef *)&constantPool[romMethodRef->classRefCPIndex]);
+			char *fullyQualifiedDefiningClassName = convertToJavaFullyQualifiedName(vmThread, definingClassFullQualifiedName);
+			char *methodSigParameters = convertMethodSignature(vmThread, J9ROMNAMEANDSIGNATURE_SIGNATURE(methodNameAndSig));
 
```

Comment:
what if one of these allocations fails?

Reply:
Good catch, `j9str_printf` represents a `NULL` value with string literal `NULL`, the thought was to still have some info when one of them fails, such as `NULL.method()` or `java.lang.String.NULL`. However it will be `NULL.NULL` when both fail. Looked it again, the messages with `NULL` seem odd as well.
Maybe just no extended message at all if either of allocation returns `NULL`?

Reply:
> Maybe just no extended message at all if either of allocation returns NULL?

Makes sense to me.  Any failure should result in no extended message (and maybe a tracepoint to help debug why)

Reply:
Agreed.
Regarding the tracepoint, existing `Trc_VM_ConvertToJavaFullyQualifiedName_Get_ClassName(vmThread, result, length, fullyQualifiedNameUTF);` can record the returned value of `convertToJavaFullyQualifiedName()`, and `Trc_VM_ConvertMethodSignature_Signature_Result(vmThread, result, bufferSize);` records  the returned value of `convertMethodSignature()`.